app/commands/warp.js

```diff
@@ -14,70 +14,220 @@ module.exports = Base => class extends Base {
+    
+                //cooldownStart goes here.
+                this.cooldownStart('warp', requestingPlayer);
+            });
+    
+            //clear requests from queue
+            this.warpRequests[playerName] = [];
+            this.tellPlayer(playerName, 'All Warp requests accepted');
+
         } else {
+            return this.tellPlayerRaw(playerName, [
+                { text: `Not a command.\n`, color: 'red' },
+                { text: `Did you mean `, color: 'white' },
+                { text: `!warp accept all`, color: 'green' },
+                { text: `?`, color: 'white' },
+            ]);
+        }
+    }
+
+    handleWarpDecline(playerName, all) {
+        //instantiate warp queue if it hasn't been instantiated yet.
+        if (!this.warpRequests[playerName]) this.warpRequests[playerName] = [];
+        
+        // if no players on the queue
+        if (this.warpRequests[playerName].length === 0) {
             this.tellPlayer(playerName, `No pending warp requests.`, 'red');
+            return;
         }
+
+        if (!all) {
+            //decline requesting player
+            let requestingPlayer = this.warpRequests[playerName][0];
+
+            //decline request and remove from array
+            this.tellPlayer(requestingPlayer, 'Warp declined');
+            this.tellPlayer(playerName, 'Warp declined');
+            this.warpRequests[playerName].shift();
+
+        } else if (all === 'all') {
+            this.warpRequests[playerName].forEach(requestingPlayer => {
+                //decline each
+                this.tellPlayer(requestingPlayer, 'Warp declined');
+            });
+    
+            //clear requests from queue
+            this.warpRequests[playerName] = [];
+            this.tellPlayer(playerName, 'All Warp requests declined');
+
+        } else {
+            return this.tellPlayerRaw(playerName, [
+                { text: `Not a command.\n`, color: 'red' },
+                { text: `Did you mean `, color: 'white' },
+                { text: `!warp decline all`, color: 'green' },
+                { text: `?`, color: 'white' },
+            ]);
+        }
+    }
+
+    handleWarpQueue(playerName) {
+        //instantiate warp queue if it hasn't been instantiated yet.
+        if (!this.warpRequests[playerName]) this.warpRequests[playerName] = [];
+
+        // if no players on the queue
+        if (this.warpRequests[playerName].length === 0) {
+            this.tellPlayer(playerName, `No pending warp requests.`, 'red');
+            return;
+        }
+
+        this.tellPlayer(playerName, 'The following players have pending warp requests:');
+        //list all players on queue
+        this.warpRequests[playerName].forEach(requestingPlayer => {
+            this.tellPlayerRaw(playerName, [
+                { text: `${requestingPlayer}`, color: 'aqua' },
+            ]);
+        });
+    }
+
+    addToQueue(acceptingPlayer, requestingPlayer) {
```

Comment:
I would rename this function to something that makes a little more sense because you're not just adding to queue

maybe `checkIsInQueueAndAdd`?